### ISSUE TITLE
Incorrect error: Invalid stage name v4-stg

### ISSUE THREAD

**Issue by ceefour** [serverless/serverless#5693]
# Serverless Frameworks forbids hyphen but API Gateway can contain hyphen

## Description

* What went wrong?

During `serverless deploy`:

> Serverless plugin "/builds/heartenly/heartenly-profile/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js" initialization errored: Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.

* What did you expect should have happened?

Deploy successful because hyphen is supported. Previously I was using ClaudiaJS with `v4-stg` stage name without any problem. Here's additional proof:

![image](https://user-images.githubusercontent.com/24123/51083354-6a01ce00-174b-11e9-8ff8-97588d706581.png)

* What was the config you used?

```yaml
provider:
  name: aws
  runtime: nodejs8.10
  stage: v4-stg
  region: us-east-1

```
* What stacktrace or error message from your provider did you see?

    Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.

Similar or dependent issues:
* #5683 

## Additional Data

* ***Serverless Framework Version you're using***: 1.36.0
* ***Operating System***: Windows 10 64-bit
* ***Stack Trace***:

    Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.

* ***Provider Error messages***:

    Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.Invalid stage name v4-stg: it should contains only [a-zA-Z0-9] for AWS provider if http event are present since API Gateway stage name cannot contains hyphens.

**Comment by gs-jackal** [comment]
I didn't use hyphen, even specifying a default stage like this fails as well
`stage: ${opt:stage, 'dev'}`

**Comment by aldarund** [comment]
same issue
  stage: ${env:STAGE}

Invalid stage name ${env:STAGE}: it should 

It seems like serverless dont interpolate stage at all

**Comment by dschep** [comment]
closed by #5686

**Comment by luanmuniz** [comment]
@dschep Shouldn't you close this issue?

**Comment by pmuens** [comment]
Closing since this shoul've been fixed via #5686.
